iothub/device/samples/PnpDeviceSamples/TemperatureController/Program.cs

```diff
@@ -0,0 +1,327 @@
+
+            bool temperatureReset = true;
+            s_maxTemp[Thermostat1] = 0d;
+            s_maxTemp[Thermostat2] = 0d;
+
+            await Task.Run(async () =>
+            {
+                while (true)
+                {
+                    if (temperatureReset)
+                    {
+                        // Generate a random value between 5.0°C and 45.0°C for the current temperature reading for each "Thermostat" component.
+                        s_temperature[Thermostat1] = Math.Round(s_random.NextDouble() * 40.0 + 5.0, 1);
+                        s_temperature[Thermostat2] = Math.Round(s_random.NextDouble() * 40.0 + 5.0, 1);
+                    }
+
+                    await SendTemperatureAsync(Thermostat1);
+                    await SendTemperatureAsync(Thermostat2);
+
+                    temperatureReset = s_temperature[Thermostat1] == 0 && s_temperature[Thermostat2] == 0;
+                    await Task.Delay(5 * 1000);
+                }
+            });
+        }
+
+        // Initialize the device client instance over Mqtt protocol (TCP, with fallback over Websocket), setting the ModelId into ClientOptions.
+        // This method also sets a connection status change callback, that will get triggered any time the device's connection status changes.
+        private static void InitializeDeviceClientAsync()
+        {
+            var options = new ClientOptions
+            {
+                ModelId = ModelId,
+            };
+            s_deviceClient = DeviceClient.CreateFromConnectionString(s_deviceConnectionString, TransportType.Mqtt, options);
+            s_deviceClient.SetConnectionStatusChangesHandler((status, reason) =>
+            {
+                s_logger.LogDebug($"Connection status change registered - status={status}, reason={reason}.");
+            });
+        }
+
+        // Send working set of device memory over telemetry.
+        private static async Task SendDeviceMemoryAsync()
+        {
+            string telemetryName = "workingSet";
+            long workingSet = Environment.WorkingSet / 1024;
+
+            Message msg = PnpHelper.CreateIothubMessageUtf8(telemetryName, JsonConvert.SerializeObject(workingSet));
+
+            await s_deviceClient.SendEventAsync(msg);
+            s_logger.LogDebug($"Telemetry: Sent - {{ \"{telemetryName}\": {workingSet}KiB }}.");
+        }
+
+        // Send device serial number over property update.
+        private static async Task SendDeviceSerialNumberAsync()
+        {
+            string propertyName = "serialNumber";
+            string propertyPatch = PnpHelper.CreatePropertyPatch(propertyName, JsonConvert.SerializeObject(SerialNumber));
+            var reportedProperties = new TwinCollection(propertyPatch);
+
+            await s_deviceClient.UpdateReportedPropertiesAsync(reportedProperties);
+            s_logger.LogDebug($"Property: Update - {{ \"{propertyName}\": \"{SerialNumber}\" }} is {StatusCode.Completed}.");
+        }
```

Comment:
Can we also get the updated format in your logger from PnP helper? This will be useful incase it changes over time. Sample maintenance will become easy.

Reply:
Do you mean - log the format of the property patches being sent? Do you want this logged, or added into the comments?

Reply:
Why not both :)

Reply:
I feel like the property formatter in `PnpHelper` is pretty descriptive in itself (it is has descriptive parameter names, and is well indented). Do you feel additional comments there are required?

Reply:
I'm more concerned about the logger.